activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -242,7 +242,16 @@ def references!(*table_names) # :nodoc:
     #   Model.select(:field).first.other_field
     #   # => ActiveModel::MissingAttributeError: missing attribute: other_field
     def select(*fields)
-      return super if block_given?
+      if block_given?
+        if fields
+          ActiveSupport::Deprecation.warn(<<-WARNING.squish)
```

Comment:
Given that "working" is "completely ignoring the supplied arguments", yes, that seems a safe bet.

(but the conditional is wrong)